src/internal/connector/onedrive/collections.go

```diff
@@ -165,6 +166,39 @@ func (c *Collections) UpdateCollections(ctx context.Context, driveID string, ite
 	return nil
 }
 
+func shouldSkipDrive(ctx context.Context, service graph.Servicer, drivePath path.Path, m folderMatcher) bool {
+	shouldSkip := !includePath(ctx, m, drivePath)
+	if shouldSkip {
+		return true
+	}
+
+	return restrictedPath(ctx, service, drivePath)
```

Comment:
```suggestion
	return !includePath(ctx, m, drivePath) || restrictedPath(ctx, service, drivePath)
```

Reply:
Resolved in [1565052](https://github.com/alcionai/corso/pull/2098/commits/1565052da72e1204d2f07bc9eb309f54bc9e3c3b)